### PR TITLE
fix: better tokenization

### PR DESCRIPTION
**Pull request by rarkins**
fix search

### CODE REVIEW

**Comment by TWiStErRob** [comment]
Wait, don't revert the whole thing, you just want to remove the case-switch tokenization.

**Comment by TWiStErRob** [comment]
```diff
-separator: '[\s\-,:!?=\[\]()<>{}"/\\]+|(?!\b)(?=[A-Z][a-z])|\.(?!\d)|&[lg]t;'
+separator: '[\s\-,:!?=\[\]()<>{}"/\\]+|\.(?!\d)|&[lg]t;'
```

**Comment by TWiStErRob** [comment]
The part `(?!\b)(?=[A-Z][a-z])` is responsible for splitting up `fooBar` into `foo` and `bar`, which gives relevant results when people don't know the whole thing.

**Comment by rarkins** [comment]
@HonkingGoose @viceice wdyt?

**Comment by HonkingGoose** [comment]
I'll defer to people that actually know regex, I just copy pasted the regex from @TWiStErRob 😄 

If you know the exact term, you should be able to find it with the search. And you should also be able to find partial matches.

File: mkdocs.yml [hunk 1]
```diff
@@ -175,8 +175,7 @@ markdown_extensions:
 
 # Plugins
 plugins:
-  - search:
-      separator: '[\s\-,:!?=\[\]()<>{}"/\\]+|(?!\b)(?=[A-Z][a-z])|\.(?!\d)|&[lg]t;'
+  - search
```
Comment (rarkins) [inline mkdocs.yml:178]:
```suggestion
  - search: '[\s\-,:!?=\[\]()<>{}"/\\]+|\.(?!\d)|&[lg]t;'
```

@TWiStErRob is this what you mean?

Comment (TWiStErRob) [reply mkdocs.yml:178]: careful with opening the yaml block
```suggestion
  - search:
      separator: '[\s\-,:!?=\[\]()<>{}"/\\]+|\.(?!\d)|&[lg]t;'
```

Comment (TWiStErRob) [reply mkdocs.yml:178]: If you commit this you should see the `(?!\b)(?=[A-Z][a-z])` part removed, along with one of the `|` (alternative operator). If you run this locally or on Codespaces you can try it out immediately.

Comment (TWiStErRob) [reply mkdocs.yml:178]: confirmed in codespaces:
image

File: mkdocs.yml [hunk 2]
```diff
@@ -176,7 +176,7 @@ markdown_extensions:
 # Plugins
 plugins:
   - search:
-      separator: '[\s\-,:!?=\[\]()<>{}"/\\]+|(?!\b)(?=[A-Z][a-z])|\.(?!\d)|&[lg]t;'
+      separator: '[\s\-,:!?=\[\]()<>{}"/\\]+|\.(?!\d)|&[lg]t;'
```
Comment (viceice) [inline mkdocs.yml:179]: @TWiStErRob is this the right thing now?

**Review by TWiStErRob — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**